webpack.config.common.js

```diff
@@ -0,0 +1,65 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path');
+const webpack = require('webpack');
+
+module.exports = {
+  entry: {
+    index: './index.js',
+    vendor: './vendor.js'
+  },
+
+  output: {
+    path: path.resolve(__dirname, './build'),
+    filename: '[name].[chunkhash].bundle.js',
+    sourceMapFilename: '[name].map',
+    chunkFilename: '[id].chunk.js'
+  },
+
+  context: path.resolve(__dirname, 'src'),
+
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        loader: 'react-hot-loader/webpack',
+        exclude: path.join(__dirname, 'node_modules')
+      },
+      {
+        test: /\.js$/,
+        enforce: 'pre',
+        loader: 'eslint-loader',
+        exclude: path.join(__dirname, 'node_modules')
+      },
+      {
+        test: /\.js$/,
+        loader: 'babel-loader',
+        query: {
+          presets: [
+            'babel-preset-env',
+            'react'
+          ]
+        },
+        exclude: path.join(__dirname, 'node_modules')
+      }
+    ]
+  },
+
+  plugins: [
+    new HtmlWebpackPlugin(
+      {
+        template: './index.html',
+        chunksSortMode: 'dependency'
+      }
+    ),
+
+    new webpack.optimize.CommonsChunkPlugin({
+      names: ['vendor', 'common'],
```

Comment:
I think we only need to define `name: common` here, right?  e.g.
```
new webpack.optimize.CommonsChunkPlugin({
  name: "common"
})
```

Otherwise by using `names` as an array means we're trying to create common chunk against
- `vendor` (good)
- `common` (doesn't exist)
- but _not_ `main` (not good).  

As per the [docs](https://webpack.js.org/plugins/commons-chunk-plugin/)
> // If an array of strings is passed this is equal to invoking the plugin multiple times for each chunk name.